### PR TITLE
Add minikube formula

### PR DESCRIPTION
**Pull request by bonifaido**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/minikube.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  version "v0.17.1"
+  sha256 "b175c355d377a6ce2fefdd19201c865a7e628581261ac949fffb725af459c389"
+
+  def install
+    mv "minikube-darwin-amd64", "minikube"
+    bin.install "minikube"
+  end
+
+  test do
+    system "minikube"
```
Comment (bfontaine) [inline Formula/minikube.rb]: Please use the full path instead: `bin/"minikube"`

File: Formula/minikube.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class Minikube < Formula
+  desc "Run Kubernetes locally"
+  homepage "https://github.com/kubernetes/minikube"
+  url "https://github.com/kubernetes/minikube/releases/download/v0.17.1/minikube-darwin-amd64"
+  version "v0.17.1"
+  sha256 "b175c355d377a6ce2fefdd19201c865a7e628581261ac949fffb725af459c389"
+
+  def install
+    mv "minikube-darwin-amd64", "minikube"
+    bin.install "minikube"
```
Comment (bfontaine) [inline Formula/minikube.rb]: Use `bin.install "foo" => "bar"` to do the equivalent of `mv "foo", "bar"; bin.install "bar"`.

File: Formula/minikube.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class Minikube < Formula
+  desc "Run Kubernetes locally"
+  homepage "https://github.com/kubernetes/minikube"
+  url "https://github.com/kubernetes/minikube/releases/download/v0.17.1/minikube-darwin-amd64"
```
Comment (bfontaine) [inline Formula/minikube.rb]: We need to build from source here.

**Review by bfontaine — CHANGES_REQUESTED** [review]

**Comment by eneveu** [comment]
Note that there is already a cask for Minikube:

```
 ~/code > brew cask info minikube
minikube: 0.18.0
https://github.com/kubernetes/minikube
Not installed
From: https://github.com/caskroom/homebrew-cask/blob/master/Casks/minikube.rb
==> Name
Minikube
==> Artifacts
minikube-darwin-amd64 (binary)
```

**Comment by bfontaine** [comment]
Let’s close this PR, then.

**Comment by commitay** [comment]
@chases Feel free to open a new PR that builds `minikube` from source.

**Comment by chases** [comment]
I notice that the Nomad formula (for example) was migrated from cask to core, would that be an acceptable solution here?

File: Formula/minikube.rb (hunk 3 above)
Comment (chases) [reply Formula/minikube.rb]: Sorry for my ignorance, couldn't we do something like `OS.mac? ? MAC_URL : LINUX_URL`. Or is there a reason to build from source other than OS compatibility?

Comment (commitay) [reply Formula/minikube.rb]: The formula must build from source.

**Pull request closed without merging**